salt/config/__init__.py

```diff
@@ -3284,6 +3292,42 @@ def _cache_id(minion_id, cache_file):
         log.error('Could not cache minion ID: {0}'.format(exc))
 
 
+def eval_minion_id_func(opts):
+    '''
+    Evaluate minion ID function if 'minion_id_type' is 'func'
+    and return the result
+    '''
+    if '__minion_id_func_evaluated' in opts:
+        return opts['id']
+
+    import salt.loader as loader
```

Comment:
That would make sense if it would be something very unique and long-loading memory consuming. But it is already imported elsewhere. Python doesn't acquire some more memory specifically for it, just creates a ref to it, which is basically nothing. So I'd avoid scattered imports.